### ISSUE TITLE
@grafana/ui#Pagination: Pagination component doesn't adjust view if clicking next to initial ellipsis

### ISSUE THREAD

**Issue by abannachGrafana** [grafana/grafana#69453]
### 
# What went wrong?

**What happened**:

  - When using the `Pagination` component if you click next to the ellipsis shown when on the first or last set of pages it does not adjust to show the next pages. See gif

![paginator-bug](https://github.com/grafana/grafana/assets/113929542/39dba0ef-5081-490f-92c7-e3830adffd21)

**What did you expect to happen**:

  - In the example gif I expect that when I click on 7 it will adjust the view and have 7 centered with 5,6,8,9 as if I were clicking the right arrow to get to 7

### 
# How do we reproduce it?

**Step 1**:

- Place a paginator component on a page that has enough pages to require the ellipsis to show

**Step 2**:

- Click on the page number immediately next to the ellipsis when on either the first or last set of pages.

### 
# What Grafana version are you using?

Grafana: 9.5.2

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

None

### Is this a Regression?

None

**Comment by joeblubaugh** [comment]
This is still an issue - I recently encountered it, and it makes browsing to a nearby page when you're near the truncation difficult.

**Comment by abannachGrafana** [comment]
Yes, I'm able to recreate on the component page https://developers.grafana.com/ui/latest/index.html?path=/story/buttons-pagination--with-pages&args=numberOfPages:20
